### PR TITLE
[skia] Update

### PR DESCRIPTION
**Pull request by dg0yt**
Based on https://github.com/microsoft/vcpkg/pull/36518, credits @abique. 

This update  makes skia use vcpkg vulkan. There are still vendored dependencies (abseil, spirv-*, maybe also dawn, graphite), but I don't want to touch that now.

### CODE REVIEW

**Comment by abique** [comment]
Shall I close the other one?

**Comment by dg0yt** [comment]
Yes, I think we can continue here.

**Comment by abique** [comment]
By the way, I wonder if the there should be a file generated which defines which features are enabled and which one aren't.

There's this file `SkUserConfig.h` but it doesn't seem to contain the build flags.

For example `SK_GRAPHITE`, `SK_VULKAN`, ...

If we have to specify those manually, the risk could be that they get out of sync which later on would turn into memory corruption due to ABI variation.

**Comment by abique** [comment]
Ah it is defined within the cmake stuff witin the `share/unofficial-skia` folder.

**Comment by dg0yt** [comment]
Yes, parts of the config are exported to CMake config.
And basically user have to request the features they want and shall rely on vcpkg's ABI tracking.

**Comment by abique** [comment]
I'm having a weird crash, the same that I had before.
I've reported it here: https://issues.skia.org/issues/324827397

**Comment by abique** [comment]
It seems that skia is built using: `-fno-exceptions -fno-rtti`.
Maybe this is causing problems?

**Comment by abique** [comment]
@dg0yt I have a small example program which uses a shader and crashes, would it be interesting to add it to the examples and check that it can build and produce an output without crashing?

**Comment by abique** [comment]
@MonicaLiu0311 should you be reviewer for this issue as it is the follow up from https://github.com/microsoft/vcpkg/pull/36518 ?

File: ports/skia/portfile.cmake [hunk 1]
```diff
@@ -26,11 +31,11 @@ declare_external_from_git(d3d12allocator
...
 declare_external_from_git(dawn
     URL "https://dawn.googlesource.com/dawn.git"
-    REF "6e25bf7674bbcb1d1e613dc0700c958830950037"
+    REF "5b45794c2c24c3fa40dc480af92c5284a95423ef"
     LICENSE_FILE LICENSE
     PATCHES
-        dawn-dedup-native-proc-gen.patch
-        add-cstdlib.patch
+        # dawn-dedup-native-proc-gen.patch
+        # add-cstdlib.patch
```
Comment (dg0yt) [inline ports/skia/portfile.cmake:38]: We still need to remove these old lines and patches.

**Comment by abique** [comment]
Hi,

Is something blocking the merge?

Cheers,
Alex

**Comment by dg0yt** [comment]
... availability of reviewers. Maybe holidays in some parts of the world.

**Review by data-queue — APPROVED** [review]

**Review by LilyWangLL — APPROVED** [review]

**Comment by abique** [comment]
@dg0yt do you know if it'd be possible to build skia with the same exception and rtti behavior as the one defined by the triplet?

**Comment by abique** [comment]
BTW, all my ABI issues with skia were coming from a specific JDK build, so this skia port is working well.
Tested in a large application with the software backend, grgl, grvk, grd3d, grmtl, graphite-vk and graphite-mtl.

**Pull request merged**